sqlglot/expressions.py

```diff
@@ -2206,7 +2206,11 @@ def build(cls, dtype: str | DataType.Type, **kwargs) -> DataType:
         from sqlglot import parse_one
 
         if isinstance(dtype, str):
-            data_type_exp = parse_one(dtype, into=DataType)
+            data_type_exp: t.Optional[Expression]
+            if dtype.upper() in cls.Type.__members__:
+                data_type_exp = DataType(this=DataType.Type[dtype.upper()])
+            else:
+                data_type_exp = parse_one(dtype, into=DataType)
```

Comment:
the build method needs to take in a dialect for the parse

Reply:
Addressed: https://github.com/tobymao/sqlglot/pull/941/commits/ae089d1eaefe86f3f8414bff4967a3578a2571aa